src/hevm/src/EVM/SymExec.hs

```diff
@@ -364,8 +367,31 @@ simplify e = if (mapExpr go e == e)
       | a == (Lit 0) = v
       | otherwise = o
     go o@(And a (And b c))
+      | a == c = (And a b)
       | a == b = (And b c)
       | otherwise = o
+    go o@(Add a b)
+      | b == (Lit 0) = a
+      | a == (Lit 0) = b
+      | otherwise = o
+    go o@(EVM.Types.LT (Lit a) (Lit b))
+      | (a < b) = Lit 1
+      | otherwise = Lit 0
+    go o@(IsZero (CallValue x))
+      | x == 0 = Lit 0x1
+      | otherwise = Lit 0x0
```

Comment:
The `x` in callvalue refers to the index of the frame for that particular instance of `CallValue`, not the value itself. We can probably tidy things up so the frame index isn't required, but as it stands this rule is wrong.

Reply:
OOOps, thanks!